src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs

```diff
@@ -23,10 +23,10 @@ public static class AWSLambdaWrapper
     private static readonly AssemblyName AssemblyName = typeof(AWSLambdaWrapper).Assembly.GetName();
 
     [SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1202:ElementsMustBeOrderedByAccess", Justification = "Initialization order.")]
-    internal static readonly string ActivitySourceName = AssemblyName.Name;
+    internal static readonly string? ActivitySourceName = AssemblyName.Name;
 
-    private static readonly Version Version = AssemblyName.Version;
-    private static readonly ActivitySource AWSLambdaActivitySource = new(ActivitySourceName, Version.ToString());
+    private static readonly Version? Version = AssemblyName.Version;
+    private static readonly ActivitySource AWSLambdaActivitySource = new(ActivitySourceName ?? "OpenTelemetry.Instrumentation.AWSLambda", Version?.ToString());
```

Comment:
`ActivitySource.ctor()` [allows the second parameter to be `null`](https://learn.microsoft.com/dotnet/api/system.diagnostics.activitysource.-ctor), so we should be able to safely ignore if we can't get a version.

Reply:
OK, it won't throw, but we still lose the version info

Reply:
Conversation should move to https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1545#discussion_r1461683945